public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -536,7 +537,17 @@ export class DashboardModel {
     for (let i = 0; i < this.panels.length; i++) {
       const panel = this.panels[i];
       if (panel.repeat) {
-        this.repeatPanel(panel, i);
+        if (!affectedPanelIds) {
+          this.repeatPanel(panel, i);
+          continue;
+        }
+
+        if (
+          affectedPanelIds.includes(panel.id) ||
+          (panel.repeatPanelId && affectedPanelIds.includes(panel.repeatPanelId))
```

Comment:
Not sure about the logic here and if `panel.repeatPanelId` is used correctly here.